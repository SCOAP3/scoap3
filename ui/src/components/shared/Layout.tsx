import React from "react";

import Footer from "./Footer";
import Header from "./Header";

interface LayoutProps {
  children: React.ReactNode;
}

const Layout: React.FC<LayoutProps> = ({ children }) => {
  return (
    <div className="app flex flex-col justify-between">
      <Header />
      <main>{children}</main>
      <Footer />
    </div>
  );
};

export default Layout;
