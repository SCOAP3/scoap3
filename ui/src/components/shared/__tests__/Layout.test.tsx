import React from "react";
import { render } from "@testing-library/react";

import Layout from "../Layout";

describe("Layout", () => {
  it("renders layout component correctly", () => {
    const { container } = render(
      <Layout>
        <div>Hi</div>
      </Layout>
    );

    expect(container).toMatchSnapshot();
  });
});
