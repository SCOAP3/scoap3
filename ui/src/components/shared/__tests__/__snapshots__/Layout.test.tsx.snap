// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders layout component correctly 1`] = `
<div>
  <div
    class="app flex flex-col justify-between"
  >
    <div
      class="header"
    >
      <div
        class="container flex items-center "
      >
        <a
          class="logo"
          href="/"
        >
          <img
            alt="Logo of Scope3"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimages%2Flogo.png&w=384&q=75"
            srcset="/_next/image?url=%2Fimages%2Flogo.png&w=256&q=75 1x, /_next/image?url=%2Fimages%2Flogo.png&w=384&q=75 2x"
            style="color: transparent;"
            width="140"
          />
        </a>
        <ul
          class="menu-desktop ant-menu collapsed"
        >
          <li
            class="ant-menu-item"
          >
            <a
              href="/"
            >
              Home
            </a>
          </li>
          <li
            class="ant-menu-item"
          >
            <a
              href="https://scoap3.org/"
              rel="noopener noreferrer"
              target="_blank"
            >
              Scoap³ project
            </a>
          </li>
          <li
            class="ant-menu-item"
          >
            <a
              href="/partners"
            >
              Partners
            </a>
          </li>
          <li
            class="ant-menu-item"
          >
            <a
              href="https://scoap3.org/scoap3-repository"
              rel="noopener noreferrer"
              target="_blank"
            >
              About
            </a>
          </li>
          <li
            class="ant-menu-item"
          >
            <a
              href="https://scoap3.org/scoap3-repository-help"
              rel="noopener noreferrer"
              target="_blank"
            >
              Help
            </a>
          </li>
          <li
            class="ant-menu-item"
          >
            <a
              href="https://github.com/SCOAP3/scoap3-next/wiki"
              rel="noopener noreferrer"
              target="_blank"
            >
              Documentation
            </a>
          </li>
          <li
            class="ant-menu-item"
          >
            <a
              href="/login"
            >
              Login
            </a>
          </li>
        </ul>
        <button
          class="ant-btn css-dev-only-do-not-override-2q8sxy ant-btn-default"
          type="button"
        >
          <span
            aria-label="menu"
            class="anticon anticon-menu"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="menu"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M904 160H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0 624H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0-312H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8z"
              />
            </svg>
          </span>
        </button>
      </div>
      <ul
        class="menu-mobile ant-menu collapsed"
      >
        <li
          class="ant-menu-item"
        >
          <a
            href="/"
          >
            Home
          </a>
        </li>
        <li
          class="ant-menu-item"
        >
          <a
            href="https://scoap3.org/"
            rel="noopener noreferrer"
            target="_blank"
          >
            Scoap³ project
          </a>
        </li>
        <li
          class="ant-menu-item"
        >
          <a
            href="/partners"
          >
            Partners
          </a>
        </li>
        <li
          class="ant-menu-item"
        >
          <a
            href="https://scoap3.org/scoap3-repository"
            rel="noopener noreferrer"
            target="_blank"
          >
            About
          </a>
        </li>
        <li
          class="ant-menu-item"
        >
          <a
            href="https://scoap3.org/scoap3-repository-help"
            rel="noopener noreferrer"
            target="_blank"
          >
            Help
          </a>
        </li>
        <li
          class="ant-menu-item"
        >
          <a
            href="https://github.com/SCOAP3/scoap3-next/wiki"
            rel="noopener noreferrer"
            target="_blank"
          >
            Documentation
          </a>
        </li>
        <li
          class="ant-menu-item"
        >
          <a
            href="/login"
          >
            Login
          </a>
        </li>
      </ul>
    </div>
    <main>
      <div>
        Hi
      </div>
    </main>
    <div
      class="footer"
    >
      <div
        class="container flex flex-col items-center"
      >
        <div
          class="container-inner flex flex-col lg:flex-row justify-between items-center lg:items-start"
        >
          <div
            class="footer-box flex-1"
          >
            <a
              href="https://scoap3.org/"
              rel="noopener noreferrer"
              target="_blank"
            >
              SCOAP3 website
            </a>
             |

            <a
              href="https://scoap3.org/scoap3-repository/"
              rel="noopener noreferrer"
              target="_blank"
            >
              About the repository
            </a>
             |

            <a
              href="https://scoap3.org/scoap3-repository/repository-help-2/"
              rel="noopener noreferrer"
              target="_blank"
            >
              Search help
            </a>
          </div>
          <div
            class="footer-box text-center"
          >
            Articles in the SCOAP3 repository are released under a

            <a
              href="https://creativecommons.org/licenses/by/3.0/"
              rel="noopener noreferrer"
              target="_blank"
            >
              CC-BY
            </a>
             license. Metadata are provided by the corresponding publishers and released under the a

            <a
              href="https://creativecommons.org/publicdomain/zero/1.0/"
              rel="noopener noreferrer"
              target="_blank"
            >
              CC0
            </a>
             waiver.
          </div>
          <div
            class="footer-box flex-1 text-right"
          >
            Repository contact:

            <a
              href="mailto:repo.admin@scoap3.org"
              rel="noopener noreferrer"
              target="_blank"
            >
              repo.admin@scoap3.org
            </a>
          </div>
        </div>
        <div>
          <p
            class="text-muted"
          >
            SCOAP3
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
